recipes/protobuf/all/test_package/conanfile.py

```diff
@@ -1,51 +1,19 @@
-import os
 from conans import ConanFile, CMake, RunEnvironment, tools
-import shutil
+import os
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
-
-    @property
-    def _protoc_available(self):
-        return not self.options["protobuf"].lite and not tools.cross_building(self.settings)
+    generators = "cmake", "cmake_find_package_multi"
 
     def build(self):
-        # Build without protoc
-        os.mkdir("without_protoc")
-        shutil.copy(os.path.join(self.source_folder, "addressbook.{}.pb.h".format(self.deps_cpp_info["protobuf"].version)),
-                    os.path.join("without_protoc", "addressbook.pb.h"))
-        shutil.copy(os.path.join(self.source_folder, "addressbook.{}.pb.cc".format(self.deps_cpp_info["protobuf"].version)),
-                    os.path.join("without_protoc", "addressbook.pb.cc"))
         cmake = CMake(self)
```

Comment:
I guess the cmake script will call the `protoc` generator.
Should the cmake build script then only be executed when not cross building?

Reply:
Yes, `protobuf_generate_cpp()` and `protobuf_generate()` are running the protoc binary

I still had no time to learn about the new cross-building way. For now, it probably is best to exclude all tests for cross-building

Reply:
Correct, current cmake script uses `cmake -E env "DYLD_LIBRARY_PATH=${Protobuf_LIB_DIRS}:${CONAN_LIB_DIRS}:${Protobuf_LIB_DIRS_RELEASE}:${Protobuf_LIB_DIRS_DEBUG}:${Protobuf_LIB_DIRS_RELWITHDEBINFO}:${Protobuf_LIB_DIRS_MINSIZEREL}" protoc ....` which is not enough to find the correct `protoc` in cross building scenarios.
Isn't this conan-io/conan#7324 required to solve this issue?

If possible I would leave cross-building for later PR. I think it's better to go step by step.

Reply:
I meant the following (make the cmake build a no-op when cross building)
```cmake
if not tools.cross_building(self.settings):
    cmake = CMake(self)
    cmake.configure()
    cmake.build()
```

Reply:
Done: https://github.com/conan-io/conan-center-index/pull/2037/commits/cb1755d7ab90bdf1a8aceccf237fbfb7876a7e2d